pkg/lib/upstreamapp/context.go

```diff
@@ -0,0 +1,35 @@
+package upstreamapp
+
+import (
+	"context"
+
+	"github.com/authgear/authgear-server/pkg/lib/session"
+)
+
+type contextKeyType struct{}
+
+var contextKey = contextKeyType{}
+
+type contextValue struct {
+	SessionInfo *session.Info
+}
+
+func WithSessionInfo(ctx context.Context, info *session.Info) context.Context {
```

Comment:
I suppose we should move `pkg/lib/session.Info` to `pkg/lib/api`, since the session headers are supposed to be public API. Then, the middleware & context can be moved there as well.